### PR TITLE
ci: Run less CI jobs by only running on relevant changes

### PR DESCRIPTION
**Pull request by rafaeelaudibert**
We're already doing this quite well, but there's some other changes we could do here.

### CODE REVIEW

File: .github/workflows/ci-e2e.yml [hunk 1]
```diff
@@ -168,7 +166,8 @@ jobs:
                   build-id: ${{ needs.container.outputs.build-id }}
                   tags: ${{ needs.container.outputs.tag }}
 
-            - name: Write .env # This step intentionally has no if, so that GH always considers the action as having run
```
Comment (rafaeelaudibert) [inline .github/workflows/ci-e2e.yml:171]: I've tested this, we don't need it, there are other required actions that are skipping every single step. We need the `action` to not have any `if`, but it's fine if all of the steps are skipped

**Review by benjackwhite — APPROVED** [review]

**Comment by daibhin** [comment]
@rafaeelaudibert we have a PR that contains only `/rust` changes: https://github.com/PostHog/posthog/pull/26796. It is not running the E2E Cypress tests. Is there any chance that was affected by this change?

**Comment by rafaeelaudibert** [comment]
@daibhin It's possible, I'm also seeing some quirks on https://github.com/PostHog/posthog/pull/26733. I'll try to revert this and see if it solves both of our problems.

**Comment by rafaeelaudibert** [comment]
@daibhin Well, that does imply that this PR affected that. I'll revert this and then eventually work on it again by slowly rolling it.

**Pull request merged**